### PR TITLE
Chore: add extension template for new methods(unify format)

### PR DESCRIPTION
**Pull request by daehyun99**
# The following checklist is mandatory.

Your PR will be closed if you remove the checklist or do not answer the questions to a satisfactory level. Use of LLM is **strictly forbidden** for any part of this checklist (even for improving language).

### Your checklist for this pull request

- [x] Have you followed all the steps from our [Contributing Guide](https://github.com/pgmpy/pgmpy/blob/dev/Contributing.md)?
- [ ] Does the PR fully address the linked issue and is within its defined scope? If you are still working on the PR, mark it as draft.
- [x] Are all the GitHub Actions checks passing? If not, mark your PR as draft while you fix it.

Please answer the following questions:

- Did you use an LLM for any assistance? Please describe how and what you used it for?
- What steps have you taken to verify that the changes correctly address the issue? And what edge cases have you considered?
- Has the LLM added try-except blocks? They will need to be removed; any error handling must be explicit.
- Have you used LLM for generating tests? They need to be compressed into a smaller number of tests without reducing coverage.

### Issue number(s) that this pull request fixes
- `None`

### List of changes to the codebase in this pull request
- Create a template to provide users with a consistent method documentation experience

### CODE REVIEW

**Comment by daehyun99** [comment]
If this is necessary, we might need to update the contributing guide as well.

File: devtools/extension_templates/_method.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# This extension template provides instructions to add new methods to pgmpy.
+#
+# Please follow the instructions below:
+# 1. Copy the `template_method` below and add it to the file you are working on.
+# 2. Implement the method's core logic and write the corresponding unit tests.
```
Comment (DARHWOLF) [inline devtools/extension_templates/_method.py]: Would it be a good idea to mention here that the tests must be `pytest` compatible, given that we are moving away from `unittets`?

Comment (daehyun99) [reply devtools/extension_templates/_method.py]: Oh, that could be confusing. I've fixed it.

**Comment by daehyun99** [comment]
I have added 'Attributes' and 'References' sections, following the structure of `_causal_discovery.py`.

Additionally, I have separated user-facing comments from engineer-facing comments.
This approach ensures that internal development notes are hidden from the official documentation and IDEs while improving overall maintainability.

**Comment by ankurankan** [comment]
@daehyun99 @daehyun99 I am a bit unsure about this. The template would depend heavily on the method that we are implementing (for example, some might require `__init__` method specification; others might not). I think the main value add here is the format of the docstring, but that is also a standard numpydoc style. What do you guys think?

**Comment by daehyun99** [comment]
@ankurankan 
That makes sense.

My suggestion was based on the idea that having a template and deleting what’s unnecessary might be a better way to prevent missing docstrings, rather than simply forgetting to write them during development.(I also considered that this process would automatically maintain the order of comments and ensure consistency.)
I thought it could be a safer approach to minimize such oversight.

If you don't think this is the right approach, feel free to close it.

Thanks.

**Comment by ankurankan** [comment]
@daehyun99, I think a better approach to achieve that would be to write a GitHub workflow that verifies that the docstring has the required sections. Most new contributors don't really look into the templates, and I think people who have been contributing for a while do not really need the template. I am, however, not sure if it is possible to test this using a github workflow and we will need to come up with a precise rule of what is acceptable.

**Comment by daehyun99** [comment]
@ankurankan 
I hadn’t considered using a GitHub workflow for documentation verifies before.
After a quick search, it seems possible to integrate `numpydoc-validation` with both `pre-commit` hooks and `GitHub workflows`.

https://numpydoc.readthedocs.io/en/latest/validation.html

Since `numpydoc` is currently only used for docstring generation, I believe extending its role to include validation would be a good approach.

https://github.com/pgmpy/pgmpy/blob/a885acf6554c5ecbf130b4e1278b0551b0fcbaf0/pyproject.toml#L66-L75

**Comment by ankurankan** [comment]
@daehyun99 That's great. Could you please create an issue for it? I will close this.
